In [1]:
!pip install sentencepiece

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [3]:
from transformers import pipeline

In [4]:
import os

In [5]:
t5 = pipeline("summarization", model="t5-base", tokenizer="t5-base", framework="tf")

/usr/local/lib/python3.10/dist-packages/huggingface_hub/utils/_token.py:72: UserWarning: 
The secret `HF_TOKEN` does not exist in your Colab secrets.
To authenticate with the Hugging Face Hub, create a token in your settings tab (https://huggingface.co/settings/tokens), set it as secret in your Google Colab and restart your session.
You will be able to reuse this secret in all of your notebooks.
Please note that authentication is recommended but still optional to access public models or datasets.
  warnings.warn(
All PyTorch model weights were used when initializing TFT5ForConditionalGeneration.

All the weights of TFT5ForConditionalGeneration were initialized from the PyTorch model.
If your task is similar to the task the model of the checkpoint was trained on, you can already use TFT5ForConditionalGeneration for predictions without further training.
/usr/local/lib/python3.10/dist-packages/transformers/models/t5/tokenization_t5_fast.py:160: FutureWarning: This tokenizer was incorrectl

In [6]:
bart = pipeline("summarization", model="facebook/bart-large-cnn", tokenizer="facebook/bart-large-cnn", framework="tf")

Some weights of the PyTorch model were not used when initializing the TF 2.0 model TFBartForConditionalGeneration: ['model.decoder.embed_tokens.weight']
- This IS expected if you are initializing TFBartForConditionalGeneration from a PyTorch model trained on another task or with another architecture (e.g. initializing a TFBertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing TFBartForConditionalGeneration from a PyTorch model that you expect to be exactly identical (e.g. initializing a TFBertForSequenceClassification model from a BertForSequenceClassification model).
Some weights or buffers of the TF 2.0 model TFBartForConditionalGeneration were not initialized from the PyTorch model and are newly initialized: ['model.shared.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.


In [7]:
pegasus = pipeline("summarization", model="google/pegasus-large", tokenizer="google/pegasus-large", framework="tf")

All model checkpoint layers were used when initializing TFPegasusForConditionalGeneration.

Some layers of TFPegasusForConditionalGeneration were not initialized from the model checkpoint at google/pegasus-large and are newly initialized: ['final_logits_bias']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.


In [8]:
# Function to read and concatenate text files
def concatenate_text_files(folder_path):
    concatenated_text = ''

    # Loop through all files in the folder
    for filename in os.listdir(folder_path):
        file_path = os.path.join(folder_path, filename)

        # Check if the file is a text file
        if os.path.isfile(file_path) and file_path.endswith('.txt'):
            with open(file_path, 'r', encoding='utf-8') as file:
                # Read the contents of the file and concatenate
                file_contents = file.read()
                concatenated_text += file_contents

    return concatenated_text

In [9]:
# folder path that contain all the text files
folder_path="/content/drive/MyDrive/Colab Notebooks/Testing Model For Summarization/Input Files"

In [10]:
# Call the function with your folder path
result_text = concatenate_text_files(folder_path)

In [11]:
# Specify the path for the output text file
output_file_path = '/content/drive/MyDrive/Colab Notebooks/Testing Model For Summarization/output_result.txt'

# Write the concatenated text to the output file
with open(output_file_path, 'w', encoding='utf-8') as output_file:
    output_file.write(result_text)

# Print a message indicating the successful write
print(f'The concatenated text has been written to {output_file_path}')


The concatenated text has been written to /content/drive/MyDrive/Colab Notebooks/Testing Model For Summarization/output_result.txt


**Input Text Data**

In [12]:
print(result_text)

History and Physical Examination:

HISTORY:
Presenting Complaints / Reasons: 
   T-cell Lymphoblastic Lymphoma
   Fever,Recurrent hypoglycaemia
   Respiratory distress and loose stools

History Of Present illness:
   T cell lymphoblastic lymphoma    
   Induction day 16
   Fever came in emergency with complain of fever for
   1 day without rigors and chills , associated with
   cough non productive .Has respiratory distress and
    recurrent hypoglycaemia
   Developed loose stools since morning multiple
   episodes as per attendant
   persistent Moaning and has weak peripheral
   pulses.CRT >2 seconds.  

Past Medical History: 
   chemo related multiple admissions

Socioeconomic History:
Marital Status:
   minor

Father's profession / Monthly income:
   dependant

Any drugs of abuse/addictions:
   none

Others:
   none

Family History: 
   insignificant

Developmental History:
   as per age

Vaccination History:
   as per epi
REVIEW OF SYSTEMS: 
General: 
   Fever + , Malaise + , Fatig

In [13]:
print(len(result_text))

69570


Totel Length of Input Text is 69570

Googlr T5 Base

In [ ]:
# Generate summary using t5
t5_summary = t5(result_text)

Token indices sequence length is longer than the specified maximum sequence length for this model (20846 > 512). Running this sequence through the model will result in indexing errors


In [15]:
print("Generated Summary Length:")
print(len(t5_summary[0]['summary_text']))

print("\nGenerated Summary:")
print(t5_summary[0]['summary_text'])